File: recursion/java/Recursion2.java
import java.util.Scanner;

public class Recursion2{

public static void printFirstNNaturalNumberReverse(int n ){
    // negative number
    if (n < 1) {
        System.out.println("Number can't be negative");
        return;
    }
    if(n == 1) {
        System.out.print(n+", ");
        return;
    }
    System.out.print(n+", ");
    printFirstNNaturalNumberReverse(n - 1);

}

public static Scanner scan = new Scanner(System.in) ;
public static void main(String ... args){
int n = scan.nextInt();
printFirstNNaturalNumberReverse(n);
}
}

File: recursion/go/Recursion2.go
package main

import "fmt"

// first n natural number
func printFirstNNaturalNumberReverse(number int) {
	// negative number
	if number < 1 {
		return
	}

	// base case
	if number == 1 {
		fmt.Print(1, ", ")
		return
	}
	// recursive call
	fmt.Print(number, ", ")
	printFirstNNaturalNumberReverse(number - 1)

}
func main() {
	var number int
	fmt.Scan(&number)
	printFirstNNaturalNumberReverse(number)
}
